### PR TITLE
Feature/cxspa 153

### PR DESCRIPTION
**Pull request by cmanciero**
https://jira.tools.sap/browse/CXSPA-153

### CODE REVIEW

File: projects/storefrontapp-e2e-cypress/cypress/helpers/product-configurator-vc.ts [hunk 1]
```diff
@@ -210,11 +210,11 @@ export function checkConflictDescriptionDisplayed(description: string): void {
 }
 
 /**
- * Navigates to the corresponding group that contains an attribute which is involved in a conflict.
+ * Navigates to the configuration group that contains an attribute which is involved in a conflict.
```
Comment (plabadie) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/product-configurator-vc.ts:213]: Why change product configuration tests?  This doesn't seem to fit the scope of the PR.
Let's discuss.

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/product_configurator/product-configurator-vc.e2e-2205-spec-flaky.ts [hunk 2]
```diff
@@ -1,98 +0,0 @@
-import { clickAllowAllFromBanner } from '../../../helpers/anonymous-consents';
```
Comment (plabadie) [inline projects/storefrontapp-e2e-cypress/cypress/integration/regression/product_configurator/product-configurator-vc.e2e-2205-spec-flaky.ts:1]: Why change this file? This doesn't seem to fit the scope of the PR.
Let's discuss.

File: projects/storefrontstyles/scss/components/product/list/_active-facets.scss [hunk 3]
```diff
@@ -14,19 +14,16 @@
       display: none;
     }
   }
-
```
Comment (plabadie) [inline projects/storefrontstyles/scss/components/product/list/_active-facets.scss:17]: Why change this file? This doesn't seem to fit the scope of the PR.
Let's discuss.

File: projects/storefrontlib/shared/components/media/media.service.ts [hunk 4]
```diff
@@ -38,7 +38,7 @@ export class MediaService {
    * for specific formats.
    */
   getMedia(
-    mediaContainer?: MediaContainer | Image,
+    mediaContainer: MediaContainer | Image,
```
Comment (plabadie) [inline projects/storefrontlib/shared/components/media/media.service.ts:41]: Why change this file? This doesn't seem to fit the scope of the PR.
Let's discuss.

File: projects/storefrontlib/shared/components/media/media.component.ts [hunk 5]
```diff
@@ -7,7 +7,6 @@ import {
   OnChanges,
   Output,
 } from '@angular/core';
-import { Image, ImageGroup } from '@spartacus/core';
```
Comment (plabadie) [inline projects/storefrontlib/shared/components/media/media.component.ts:10]: Why change this file? This doesn't seem to fit the scope of the PR.
Let's discuss.

File: projects/storefrontlib/cms-components/product/product-list/product-facet-navigation/active-facets/active-facets.component.ts [hunk 6]
```diff
@@ -42,15 +42,4 @@ export class ActiveFacetsComponent {
       ? ''
       : facet.facetValueName;
   }
-
```
Comment (plabadie) [inline projects/storefrontlib/cms-components/product/product-list/product-facet-navigation/active-facets/active-facets.component.ts:45]: Why change this file? This doesn't seem to fit the scope of the PR.
Let's discuss.

File: projects/storefrontlib/cms-components/product/product-list/product-facet-navigation/active-facets/active-facets.component.spec.ts [hunk 7]
```diff
@@ -9,7 +9,7 @@ import { By } from '@angular/platform-browser';
 import { RouterTestingModule } from '@angular/router/testing';
 import { Breadcrumb, I18nTestingModule } from '@spartacus/core';
 import { KeyboardFocusModule } from '../../../../../layout/a11y/keyboard-focus/keyboard-focus.module';
-import { EMPTY, of } from 'rxjs';
```
Comment (plabadie) [inline projects/storefrontlib/cms-components/product/product-list/product-facet-navigation/active-facets/active-facets.component.spec.ts:12]: Why change this file? This doesn't seem to fit the scope of the PR.
Let's discuss.

File: projects/storefrontlib/cms-components/product/product-list/product-facet-navigation/active-facets/active-facets.component.html [hunk 8]
```diff
@@ -8,11 +8,6 @@ <h4 *ngIf="facetList?.activeFacets?.length > 0">
     routerLink="./"
     [queryParams]="getLinkParams(facet)"
     [cxFocus]="getFocusKey(facetList, facet)"
-    role="button"
```
Comment (plabadie) [inline projects/storefrontlib/cms-components/product/product-list/product-facet-navigation/active-facets/active-facets.component.html:11]: Why change this file? This doesn't seem to fit the scope of the PR.
Let's discuss.

File: feature-libs/asm/components/customer-emulation/customer-emulation.component.html [hunk 9]
```diff
@@ -1,11 +1,73 @@
...
+        ></cx-icon>
+        {{ 'asm.assignCart.error' | cxTranslate }}</label
+      >
+    </div>
+    <button
+      class="assignCartToCustomer"
+      [disabled]="!cartIdExists"
+      (click)="assignCartToCustomer()"
+      [class.active]="cartIdExists"
+      data-cy="assignCart"
```
Comment (plabadie) [inline feature-libs/asm/components/customer-emulation/customer-emulation.component.html]: About `data-cy`:  The storefront markup should be free of any test-related attributes.  Please remove `data-cy` and similar attributes for test usage only.
Instead, we can use simple css selectors.
So far, it was possible to select the right element in unit tests and e2es using this technique.

File: feature-libs/asm/components/asm-main-ui/asm-main-ui.component.html [hunk 10]
```diff
@@ -43,7 +43,10 @@
     *ngIf="customerSupportAgentLoggedIn$ | async; else showLoginForm"
   >
     <ng-container *ngIf="customer$ | async; else showCustomerSelection">
-      <cx-customer-emulation *ngIf="notCollapsed"></cx-customer-emulation>
+      <cx-customer-emulation
+        *ngIf="notCollapsed"
+        (submitEvent)="startCustomerEmulationSession($event)"
```
Comment (plabadie) [inline feature-libs/asm/components/asm-main-ui/asm-main-ui.component.html:48]: Why is `(submitEvent)="startCustomerEmulationSession($event)"` added and which component is listening for it?

File: feature-libs/asm/occ/adapters/default-occ-asm-config.ts [hunk 11]
```diff
@@ -5,6 +5,7 @@ export const defaultOccAsmConfig: OccConfig = {
     occ: {
       endpoints: {
         asmCustomerSearch: '/assistedservicewebservices/customers/search',
+        asmBindCart: '/assistedservicewebservices/customers/search',
```
Comment (plabadie) [inline feature-libs/asm/occ/adapters/default-occ-asm-config.ts]: This is not the right url for asmBindCart

File: feature-libs/asm/core/connectors/asm.adapter.ts [hunk 12]
```diff
@@ -11,4 +11,6 @@ export abstract class AsmAdapter {
   abstract customerSearch(
     options: CustomerSearchOptions
   ): Observable<CustomerSearchPage>;
+
+  abstract bindCart(cartId: string, customerId: string): Observable<void>;
```
Comment (plabadie) [inline feature-libs/asm/core/connectors/asm.adapter.ts:15]: `Observable<void>` should probably be something else than void.  Let's look at other similar functions to see what type is used in similar cases.

File: feature-libs/asm/occ/adapters/occ-asm.adapter.ts [hunk 13]
```diff
@@ -65,4 +65,27 @@ export class OccAsmAdapter implements AsmAdapter {
...
+      true,
+      new HttpHeaders()
+    );
+    let params: HttpParams = new HttpParams()
+      .set('baseSite', this.activeBaseSite)
+      .set('cartId', cartId)
+      .set('customerId', customerId);
+
+    const url = this.occEndpointsService.buildUrl(
+      'assistedservicewebservices/bind-cart',
```
Comment (plabadie) [inline feature-libs/asm/occ/adapters/occ-asm.adapter.ts]: Url should not be hardcoded here, but instead taken from the endpoint config.  There are examples for this in other occ adapters.

File: projects/core/src/i18n/index.ts [hunk 14]
```diff
@@ -1,5 +1,4 @@
 export * from './config/i18n-config';
-export * from './config/i18n-config-initializer';
```
Comment (plabadie) [inline projects/core/src/i18n/index.ts:2]: Why?

File: projects/assets/src/translations/en/product.ts [hunk 15]
```diff
@@ -17,8 +17,6 @@ export const product = {
       label: 'Filter by',
       action: 'Filter by',
     },
-    activeFilter:
-      '{{filter}} filter, clicking on this button will remove the filter',
```
Comment (plabadie) [inline projects/assets/src/translations/en/product.ts:21]: Is this used in the new asm feature?

File: projects/storefrontapp-e2e-cypress/cypress/helpers/asm.ts [hunk 16]
```diff
@@ -168,3 +177,11 @@ export function testCustomerEmulation() {
     cy.get('cx-asm-main-ui').should('not.be.visible');
   });
 }
+
+export function bindCart() {
+  const bindingRequest = listenForCartBindingRequest();
+  //click button
+  cy.get('[data-cy="assignCart"]').click();
```
Comment (plabadie) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/asm.ts]: See other comment about `data-cy`

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/asm/asm.e2e-spec.ts [hunk 17]
```diff
@@ -71,7 +72,68 @@ context('Assisted Service Module', () => {
...
+      cy.get('cx-asm-main-ui').should('not.be.visible');
+    });
+
+    it('agent should be able to bind anonymous cart to customer', () => {
+      let assignedCartId: string;
+      checkout.visitHomePage();
+      cy.get('cx-asm-main-ui').should('not.exist');
+
+      cy.log('--> Add to cart as an anonymous user');
+      cart.addProductAsAnonymous();
```
Comment (plabadie) [inline projects/storefrontapp-e2e-cypress/cypress/integration/regression/asm/asm.e2e-spec.ts:93]: Perhaps we can create an anonymous cart with api calls.  Just an idea. To be explored/discussed.

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/asm/asm.e2e-spec.ts [hunk 18]
```diff
@@ -71,7 +72,68 @@ context('Assisted Service Module', () => {
...
+        .get('h2.cx-total')
+        .then(($cartId) => {
+          let cartId: string;
+          const text = $cartId.text();
+          cartId = text.replace('Cart #', '').trim();
+
+          cy.log('--> Agent logging in');
+          checkout.visitHomePage('asm=true');
+          cy.get('cx-asm-main-ui').should('exist');
+          cy.get('cx-asm-main-ui').should('be.visible');
```
Comment (plabadie) [inline projects/storefrontapp-e2e-cypress/cypress/integration/regression/asm/asm.e2e-spec.ts:106]: These assessments have been done before I think.

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/asm/asm.e2e-spec.ts [hunk 19]
```diff
@@ -71,7 +72,68 @@ context('Assisted Service Module', () => {
...
+          asm.agentLogin();
+
+          cy.log('--> Starting customer emulation');
+          asm.startCustomerEmulation(customer);
+
+          assignedCartId = cartId;
```
Comment (plabadie) [inline projects/storefrontapp-e2e-cypress/cypress/integration/regression/asm/asm.e2e-spec.ts]: Can cartId be used directly?  It doesn't seem re-assigned.

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/asm/asm.e2e-spec.ts [hunk 20]
```diff
@@ -71,7 +72,68 @@ context('Assisted Service Module', () => {
...
+      cy.get('cx-cart-details')
+        .get('h2.cx-total')
+        .then(($cartId) => {
+          expect($cartId.text()).to.contain(`Cart #${assignedCartId}`);
+        });
+
+      cy.log(
+        '--> Stop customer emulation using the end session button in the ASM UI'
+      );
+      cy.get('cx-customer-emulation [data-cy=logout]').click();
```
Comment (plabadie) [inline projects/storefrontapp-e2e-cypress/cypress/integration/regression/asm/asm.e2e-spec.ts:133]: Potentially just `asm.agentSignOut();` is enough in the end (lines 136-145).  Let's discuss.

File: projects/schematics/src/dependencies.json [hunk 21]
```diff
@@ -55,8 +55,10 @@
     "@angular/common": "^13.3.0",
     "@angular/core": "^13.3.0",
     "@angular/forms": "^13.3.0",
+    "@ng-select/ng-select": "^8.0.0",
     "@ngrx/effects": "^13.0.2",
     "@ngrx/store": "^13.0.2",
+    "@spartacus/cart": "4.1.0-next.0",
```
Comment (plabadie) [inline projects/schematics/src/dependencies.json:61]: Let's discuss to see if there is an alternative to adding a dependency to the cart in ASM

File: feature-libs/asm/package.json [hunk 22]
```diff
@@ -29,8 +24,10 @@
     "@angular/common": "^13.3.0",
     "@angular/core": "^13.3.0",
     "@angular/forms": "^13.3.0",
+    "@ng-select/ng-select": "^8.0.0",
     "@ngrx/effects": "^13.0.2",
     "@ngrx/store": "^13.0.2",
+    "@spartacus/cart": "4.1.0-next.0",
```
Comment (plabadie) [inline feature-libs/asm/package.json:30]: Let's discuss to see if there is an alternative to adding a dependency to the cart in ASM

File: feature-libs/asm/assets/translations/en/asm.ts [hunk 23]
```diff
@@ -26,6 +26,31 @@ export const asm = {
...
+        view: '360 VIEW',
+      },
+      tableSort: {
+        dateSortDesc: 'Date(Desc)',
+        dateSortAsc: 'Date(Asc)',
+        nameSortDesc: 'Name(Desc)',
+        nameSortAsc: 'Name(Asc)',
+      },
+      noOfCustomers: '{{count}} Customers',
+      noCustomers: 'No customer found.',
```
Comment (plabadie) [inline feature-libs/asm/assets/translations/en/asm.ts]: These labels don't seem related to cart binding.

File: feature-libs/asm/components/customer-emulation/customer-emulation.component.ts [hunk 24]
```diff
@@ -1,20 +1,40 @@
...
   protected subscription = new Subscription();
 
+  @Output()
+  submitEvent = new EventEmitter<{ customerId?: string }>();
+
   constructor(
     protected asmComponentService: AsmComponentService,
-    protected userService: UserService
+    protected userService: UserService,
+    protected occAsmAdapter: OccAsmAdapter,
```
Comment (plabadie) [inline feature-libs/asm/components/customer-emulation/customer-emulation.component.ts:36]: A component can not depend directly on an OCC adapter and make http call itself.
Whatever logic required to query the backend should be encapsulated by a facade service

File: feature-libs/asm/components/customer-emulation/customer-emulation.component.ts [hunk 25]
```diff
@@ -25,12 +45,54 @@ export class CustomerEmulationComponent implements OnInit, OnDestroy {
...
+
+    this.cartId.valueChanges.subscribe((response: string) => {
+      this.cartIdExists = response.trim().length > 0;
+    });
+
+    this.activeCartFacade.getActiveCartId().subscribe((response) => {
+      if (response) {
+        this.cartId.setValue(response);
+      }
+    });
```
Comment (plabadie) [inline feature-libs/asm/components/customer-emulation/customer-emulation.component.ts:57]: I would like to understand the purpose of the cart facade logic here.  Cart binding is done by the asm endpoint.  Let's discuss.

File: feature-libs/asm/components/customer-emulation/customer-emulation.component.ts [hunk 26]
```diff
@@ -25,12 +45,54 @@ export class CustomerEmulationComponent implements OnInit, OnDestroy {
...
+   * Assign the input cart number to the customer
+   */
+  assignCartToCustomer() {
+    const customerId = this.customer.uid;
+    const cartId = this.cartId.value;
+
+    if (customerId) {
+      this.occAsmAdapter.bindCart(cartId, customerId).subscribe(
+        () => {
+          this.showAssignCartSuccess = true;
```
Comment (plabadie) [inline feature-libs/asm/components/customer-emulation/customer-emulation.component.ts:74]: There is a standard way of displaying success and error messages.  It would be more consistent to use them.  Let's discuss.

File: feature-libs/asm/components/customer-emulation/customer-emulation.component.ts [hunk 27]
```diff
@@ -25,12 +45,54 @@ export class CustomerEmulationComponent implements OnInit, OnDestroy {
...
 
+  /**
+   * Assign the input cart number to the customer
+   */
+  assignCartToCustomer() {
+    const customerId = this.customer.uid;
+    const cartId = this.cartId.value;
+
+    if (customerId) {
+      this.occAsmAdapter.bindCart(cartId, customerId).subscribe(
```
Comment (plabadie) [inline feature-libs/asm/components/customer-emulation/customer-emulation.component.ts:72]: As mentioned earlier, occ calls can't be done from the component.

File: feature-libs/asm/components/customer-emulation/customer-emulation.component.ts [hunk 28]
```diff
@@ -1,20 +1,40 @@
-import { Component, OnDestroy, OnInit } from '@angular/core';
+import {
```
Comment (plabadie) [inline feature-libs/asm/components/customer-emulation/customer-emulation.component.ts:1]: It would probably improve the design to encapsulate the cart binding functionality in its own dedicated component.  Let's discuss.

File: feature-libs/asm/styles/components/_asm-main-ui.component.scss [hunk 29]
```diff
@@ -19,7 +19,7 @@
   }
 
   .logout {
-    background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='16' height='16' viewBox='0 0 16 16'%3E%3Cpath fill='%23d1e3ff' d='M11,2.7c1.2,0.6,2.2,1.5,2.9,2.6c1.3,1.9,1.5,4.4,0.6,6.5c-0.3,0.8-0.8,1.6-1.5,2.2c-0.6,0.6-1.4,1.1-2.2,1.5 C9.9,15.8,9,16,8,16c-0.9,0-1.9-0.2-2.7-0.5c-0.8-0.4-1.6-0.9-2.2-1.5c-0.6-0.6-1.1-1.4-1.5-2.2C0.7,9.6,0.9,7.2,2.1,5.3 c0.7-1.1,1.7-2,2.9-2.6v1.1C4.1,4.3,3.3,5.1,2.8,6C2.3,6.9,2,7.9,2,9c0,1.6,0.6,3.2,1.8,4.3c0.5,0.5,1.2,1,1.9,1.3 c1.5,0.6,3.2,0.6,4.7,0c0.7-0.3,1.4-0.7,1.9-1.3C13.4,12.1,14,10.6,14,9c0-1.1-0.3-2.1-0.8-3c-0.5-0.9-1.3-1.7-2.2-2.2 C11,3.8,11,2.7,11,2.7z M8,9C7.7,9,7.5,8.9,7.3,8.7C7.1,8.5,7,8.3,7,8V1c0-0.3,0.1-0.5,0.3-0.7c0.4-0.4,1-0.4,1.4,0 C8.9,0.5,9,0.7,9,1v7c0,0.3-0.1,0.5-0.3,0.7C8.5,8.9,8.2,9,8,9z'/%3E%3C/svg%3E%0A");
```
Comment (plabadie) [inline feature-libs/asm/styles/components/_asm-main-ui.component.scss:22]: Why update the rest of the ASM's styling within this PR?  This doesn't seem to be in scope of the cart binding ticket.  Let's discuss.


Same comment applies for all the sass changes below.

File: feature-libs/asm/components/customer-emulation/customer-emulation.component.ts [hunk 30]
```diff
@@ -25,12 +45,54 @@ export class CustomerEmulationComponent implements OnInit, OnDestroy {
...
+  assignCartToCustomer() {
+    const customerId = this.customer.uid;
+    const cartId = this.cartId.value;
+
+    if (customerId) {
+      this.occAsmAdapter.bindCart(cartId, customerId).subscribe(
+        () => {
+          this.showAssignCartSuccess = true;
+
+          setTimeout(() => {
```
Comment (plabadie) [inline feature-libs/asm/components/customer-emulation/customer-emulation.component.ts:76]: Is there logic somewhere to reload the cart in the storefront?  We probably need that somehow.  Not sure, but worth looking into.

File: feature-libs/asm/components/customer-emulation/customer-emulation.component.ts [hunk 31]
```diff
@@ -25,12 +45,54 @@ export class CustomerEmulationComponent implements OnInit, OnDestroy {
...
+            this.showAssignCartSuccess = false;
+            this.submitEvent.emit({ customerId: customerId });
+          }, 3000);
+        },
+        () => {
+          this.showAssignCartError = true;
```
Comment (plabadie) [inline feature-libs/asm/components/customer-emulation/customer-emulation.component.ts:82]: There seems to be a bug in the case when the cart binding is refused by the backend. 
If I do so, and sign out completely the cs agent, I can't add anything to the cart once I'm back in the regular anon customer session.

File: feature-libs/asm/components/customer-emulation/customer-emulation.component.ts [hunk 32]
```diff
@@ -25,12 +45,54 @@ export class CustomerEmulationComponent implements OnInit, OnDestroy {
...
+
```
Comment (plabadie) [inline feature-libs/asm/components/customer-emulation/customer-emulation.component.ts:83]: Other behaviour question: there is a cart id pre-populated in the cart binding input field when it's displayed.  Where does that come from and what was the intent?  (also, it resulted in an error when I tried to bind it.)

File: feature-libs/asm/components/customer-emulation/customer-emulation.component.spec.ts [hunk 33]
```diff
@@ -74,11 +104,67 @@ describe('CustomerEmulationComponent', () => {
...
+    expect(component.cartIdExists).toBeTruthy();
+
+    // check that cart id entered matches
+    expect(component.cartId.value).toEqual(testCartId);
+
+    // check that assign to cart button is enabled
+    expect(
+      assignCartToCustomerButton.nativeElement.getAttribute('disabled')
+    ).toBeFalsy();
+  });
```
Comment (plabadie) [inline feature-libs/asm/components/customer-emulation/customer-emulation.component.spec.ts:169]: At fist glance, the unit test seems to be written like an e2e test.  Can the design be adjusted to test individual behaviours? (like the component doesn't submit the form if the field is empty.). Perhaps what it needs is more coverage.  I'm just giving suggestions.

File: feature-libs/asm/components/customer-emulation/customer-emulation.component.ts (hunk 32 above)
Comment (cmanciero) [reply feature-libs/asm/components/customer-emulation/customer-emulation.component.ts:83]: The cart id that is pre-populated is the same functionality as Accelerator. It displays the current assigned cart for the selected customer. It fails because the cart is already assigned to a customer.

File: feature-libs/asm/styles/components/_asm-main-ui.component.scss (hunk 29 above)
Comment (cmanciero) [reply feature-libs/asm/styles/components/_asm-main-ui.component.scss:22]: The ASM styling needs to be updated because the ticket asks for new functionality and style updates

File: projects/assets/src/translations/en/product.ts (hunk 15 above)
Comment (cmanciero) [reply projects/assets/src/translations/en/product.ts:21]: Don't know why this is here. I didn't make any changes to this file

File: projects/core/src/i18n/index.ts (hunk 14 above)
Comment (cmanciero) [reply projects/core/src/i18n/index.ts:2]: Don't know why this is here. I didn't make any changes to this file. Maybe during a /develop branch merge it changed these files

File: projects/storefrontapp-e2e-cypress/cypress/helpers/product-configurator-vc.ts (hunk 1 above)
Comment (cmanciero) [reply projects/storefrontapp-e2e-cypress/cypress/helpers/product-configurator-vc.ts:213]: Don't know why this is here. I didn't make any changes to this file. Maybe during a /develop branch merge it changed these files

File: projects/storefrontlib/cms-components/product/product-list/product-facet-navigation/active-facets/active-facets.component.html (hunk 8 above)
Comment (cmanciero) [reply projects/storefrontlib/cms-components/product/product-list/product-facet-navigation/active-facets/active-facets.component.html:11]: Don't know why this is here. I didn't make any changes to this file. Maybe during a /develop branch merge it changed these files

File: projects/storefrontlib/cms-components/product/product-list/product-facet-navigation/active-facets/active-facets.component.spec.ts (hunk 7 above)
Comment (cmanciero) [reply projects/storefrontlib/cms-components/product/product-list/product-facet-navigation/active-facets/active-facets.component.spec.ts:12]: Don't know why this is here. I didn't make any changes to this file. Maybe during a /develop branch merge it changed these files

File: projects/storefrontlib/cms-components/product/product-list/product-facet-navigation/active-facets/active-facets.component.ts (hunk 6 above)
Comment (cmanciero) [reply projects/storefrontlib/cms-components/product/product-list/product-facet-navigation/active-facets/active-facets.component.ts:45]: Don't know why this is here. I didn't make any changes to this file. Maybe during a /develop branch merge it changed these files

File: projects/storefrontlib/shared/components/media/media.component.ts (hunk 5 above)
Comment (cmanciero) [reply projects/storefrontlib/shared/components/media/media.component.ts:10]: Don't know why this is here. I didn't make any changes to this file. Maybe during a /develop branch merge it changed these files

File: projects/storefrontlib/shared/components/media/media.service.ts (hunk 4 above)
Comment (cmanciero) [reply projects/storefrontlib/shared/components/media/media.service.ts:41]: Don't know why this is here. I didn't make any changes to this file. Maybe during a /develop branch merge it changed these files

File: projects/storefrontstyles/scss/components/product/list/_active-facets.scss (hunk 3 above)
Comment (cmanciero) [reply projects/storefrontstyles/scss/components/product/list/_active-facets.scss:17]: Don't know why this is here. I didn't make any changes to this file. Maybe during a /develop branch merge it changed these files

File: feature-libs/asm/components/asm-main-ui/asm-main-ui.component.html (hunk 10 above)
Comment (cmanciero) [reply feature-libs/asm/components/asm-main-ui/asm-main-ui.component.html:48]: The asm-main component is listening for when the cart is assigned to the selected customer so it can refresh the customer session with the selected cart.

File: feature-libs/asm/components/customer-emulation/customer-emulation.component.ts (hunk 25 above)
Comment (cmanciero) [reply feature-libs/asm/components/customer-emulation/customer-emulation.component.ts:57]: I will recheck to see if this functionality is needed

File: feature-libs/asm/components/customer-emulation/customer-emulation.component.ts (hunk 26 above)
Comment (cmanciero) [reply feature-libs/asm/components/customer-emulation/customer-emulation.component.ts:74]: Discuss with UX on using existing Spartacus toast messages

File: feature-libs/asm/components/customer-emulation/customer-emulation.component.ts (hunk 30 above)
Comment (cmanciero) [reply feature-libs/asm/components/customer-emulation/customer-emulation.component.ts:76]: Will look into using method to reload cart instead of re-login customer

File: projects/storefrontlib/cms-components/product/product-list/product-facet-navigation/active-facets/active-facets.component.html (hunk 8 above)
Comment (cmanciero) [reply projects/storefrontlib/cms-components/product/product-list/product-facet-navigation/active-facets/active-facets.component.html:11]: In an earlier commit when merging a branch to this one, these changes existed. The revert of that commit is undoing those changes.

File: projects/storefrontlib/cms-components/product/product-list/product-facet-navigation/active-facets/active-facets.component.ts (hunk 6 above)
Comment (cmanciero) [reply projects/storefrontlib/cms-components/product/product-list/product-facet-navigation/active-facets/active-facets.component.ts:45]: In an earlier commit when merging a branch to this one, these changes existed. The revert of that commit is undoing those chang

File: projects/storefrontlib/shared/components/media/media.component.ts (hunk 5 above)
Comment (cmanciero) [reply projects/storefrontlib/shared/components/media/media.component.ts:10]: In an earlier commit when merging a branch to this one, these changes existed. The revert of that commit is undoing those chang

File: projects/storefrontlib/shared/components/media/media.service.ts (hunk 4 above)
Comment (cmanciero) [reply projects/storefrontlib/shared/components/media/media.service.ts:41]: In an earlier commit when merging a branch to this one, these changes existed. The revert of that commit is undoing those changes

File: projects/storefrontstyles/scss/components/product/list/_active-facets.scss (hunk 3 above)
Comment (cmanciero) [reply projects/storefrontstyles/scss/components/product/list/_active-facets.scss:17]: In an earlier commit when merging a branch to this one, these changes existed. The revert of that commit is undoing those changes

File: feature-libs/asm/components/customer-emulation/customer-emulation.component.ts (hunk 31 above)
Comment (plabadie) [reply feature-libs/asm/components/customer-emulation/customer-emulation.component.ts:82]: The issue seems resolved.
